### PR TITLE
Add large file download warning and add AzCopy example to upload warning

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

image

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by nthandeMS — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py [hunk 1]
```diff
@@ -90,8 +91,9 @@ def upload(
             # warn if large file (> 100 MB)
             file_size, _ = get_directory_size(source)
             file_size_in_mb = file_size / 10**6
+            full_storage_url = f"https://{self.account_name}.blob.core.windows.net/{self.container}/{dest}"
```
Comment (harneetvirk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py:94]: .blob.core.windows.net won't work in non public cloud.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py [hunk 2]
```diff
@@ -221,6 +224,13 @@ def download(
                     continue
 
                 blob_content = self.container_client.download_blob(item)
+
+                # check if total size of download has exceeded 100 MB
+                full_storage_url = f"https://{self.account_name}.blob.core.windows.net/{self.container}/{starts_with}"
```
Comment (harneetvirk) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py:229]: Same issue of hard coded blob endpoint for public cloud.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py:94]: What would the change need to be for private clouds? If you have a short explanation, I can add a note to the message.

Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py]: Addressed offline and fixed in code.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_blob_storage_helper.py]: Addressed offline and fixed in code.

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**